### ISSUE TITLE
Port 80 is hard-coded in http.py

### ISSUE THREAD

**Issue by willfroning** [secdev/scapy#2461]
When attempting to create a `http_request()` with a port, it is only created for `port=80`.

I've created a small patch in my repo available here: https://github.com/willfroning/scapy/commit/c07d903ccf86039fd6e0e91df0b604ad68e919a6
```diff
    }
    http_headers.update(headers)
    req = HTTP() / HTTPRequest(**http_headers)
-   tcp_client = TCP_client.tcplink(HTTP, host, 80)
+   tcp_client = TCP_client.tcplink(HTTP, host, port)
    ans = None
    try:
        ans = tcp_client.sr1(req, timeout=timeout, verbose=verbose)
```

**Comment by guedou** [comment]
Thanks. Can you create a pull-request with this fix?

**Comment by p-l-** [comment]
Fixed by #2462.
